api/app/utils.py

```
@@ -0,0 +1,60 @@
+def download_file_from_r2(file_key: str) -> bytes:
+    s3 = get_s3_client()
+    try:
+        response = s3.get_object(Bucket=settings.r2_bucket_name, Key=file_key)
+        file_content = response['Body'].read()
+        return file_content
+    except ClientError as e:
+        if e.response['Error']['Code'] == 'NoSuchKey':
+            raise HTTPException(
+                status_code=status.HTTP_404_NOT_FOUND,
+                detail=f"File not found in R2 bucket: {file_key}"
+            )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)